astropy/units/core.py

```diff
@@ -1196,10 +1196,7 @@ def _compose(
         def is_final_result(unit):
             # Returns True if this result contains only the expected
             # units
-            for base in unit.bases:
-                if base not in namespace:
-                    return False
-            return True
+            return all(base in namespace for base in unit.bases)
```

Comment:
thinking about this again, this might be even faster as `set(unit.bases).issubset(namespace)`. But probably good not to change the code beyond the obvious in this PR!

Reply:
Might be faster, but yeah, for a future PR.